packages/gamut/src/Form/SelectDropdown.tsx

```diff
@@ -99,67 +139,67 @@ const customStyles: StylesConfig<OptionTypeBase, false> = {
   }),
 };
 
-const ChevronDropdown = (props: IndicatorProps<OptionTypeBase, false>) => {
-  return (
-    <DropdownIndicator {...props}>
-      <ArrowChevronDownIcon size={16} />
-    </DropdownIndicator>
-  );
+const defaultProps = {
+  name: undefined,
+  isSearchable: false,
+  isMulti: false,
+  styles: customStyles,
+  components: {
+    DropdownIndicator: ChevronDropdown,
+    IndicatorSeparator: () => null,
+    SelectContainer: CustomContainer,
+  },
 };
 
 export const SelectDropdown: React.FC<SelectDropdownProps> = ({
   options,
   error,
   id,
   disabled,
-  defaultValue,
+  onChange,
+  value = undefined,
+  name,
+  placeholder = 'Select an option',
+  inputProps,
   ...rest
 }) => {
   const [activated, setActivated] = useState(false);
+  const baseInputProps = { name };
 
   const changeHandler = (optionEvent: OptionStrict) => {
-    rest?.onChange?.(optionEvent, {
+    onChange?.(optionEvent, {
       action: 'select-option',
       option: optionEvent,
     });
     setActivated(true);
   };
 
-  const selectOptions: Array<OptionTypeBase> = [];
+  const selectOptions = useMemo(() => {
+    return parseOptions({ options, id, selectDropdown: true });
+  }, [options, id]);
 
-  if (options instanceof Array) {
-    options.forEach((option) => {
-      const key = id ? `${id}-${option}` : option;
-      selectOptions.push({ label: key, value: option });
-    });
-  } else if (isObject(options)) {
-    each(options, (text, val) => {
-      selectOptions.push({ label: text, value: val });
-    });
-  }
+  const parsedValue = useMemo(() => {
+    const currentValue = selectOptions.find(
+      ({ value: optionValue }) => optionValue === value
+    );
 
-  const setDefaultValue = rest.placeholder
-    ? null
-    : defaultValue
-    ? selectOptions.find((option) => option.value === defaultValue)
-    : selectOptions[0];
+    return currentValue;
+  }, [selectOptions, value]);
 
   return (
     <ReactSelect
+      {...defaultProps}
       id={id || rest.htmlFor}
-      defaultValue={setDefaultValue}
-      styles={customStyles}
+      value={parsedValue}
       activated={activated}
       error={Boolean(error)}
-      components={{
-        DropdownIndicator: ChevronDropdown,
-        IndicatorSeparator: () => null,
-      }}
       onChange={changeHandler}
-      isSearchable={false}
-      isMulti={false}
+      inputProps={
+        inputProps ? { ...inputProps, ...baseInputProps } : baseInputProps
+      }
```

Comment:
Object spread works with undefined values :) 
```suggestion
      inputProps={{ ...inputProps, ...baseInputProps }}
```